app/services/google_maps/records_parser.rb

```diff
@@ -35,10 +35,25 @@ def parse_json(json)
     {
       latitude: json['latitudeE7'].to_f / 10**7,
       longitude: json['longitudeE7'].to_f / 10**7,
-      timestamp: DateTime.parse(json['timestamp']).to_i,
+      timestamp: parse_timestamp(json['timestamp'] || json['timestampMs']),
       altitude: json['altitude'],
       velocity: json['velocity'],
       raw_data: json
     }
   end
+
+  def parse_timestamp(timestamp)
```

Comment:
This method is the same across `app/services/google_maps/records_parser.rb` and `app/services/google_maps/semantic_history_parser.rb`. Can you please move it to a module so it wouldn't be repeated?

Also tests for both cases (with `timestamp` and `timestampMs`) would be great to have.

Reply:
Master branch now has tests for both classes changed in this PR, so you can take them as an example

Reply:
Code defined in helpers by default is only available in views, not in service classes. I'd place it in module somewhere in /lib/timestamps.rb, for example